plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php

```diff
@@ -650,13 +692,13 @@ protected function getDirectlyRelatedTerms($typeId)
     $sql .= ' FROM '.QubitObjectTermRelation::TABLE_NAME.' otr';
     $sql .= ' JOIN '.QubitTerm::TABLE_NAME.' current
                 ON otr.term_id = current.id';
-    $sql .= ' WHERE otr.object_id = ?
-                AND current.taxonomy_id = ?';
+    $sql .= ' WHERE otr.object_id = :id
+                AND current.taxonomy_id = :taxonomyId';
 
     self::$statements['relatedTerms'] = self::$conn->prepare($sql);
-    self::$statements['relatedTerms']->execute(array($this->__get('id'), $typeId));
+    self::$statements['relatedTerms']->execute(array(':id' => $this->__get('id'), ':taxonomyId' => $typeId));
 
-    return self::$statements['relatedTerms']->fetchAll(PDO::FETCH_OBJ);
+    return self::$statements['relatedTerms']->fetchAll(PDO::FETCH_ASSOC);
```

Comment:
The `getDirectlyRelatedTerms` and `getRelatedTerms` functions are now pretty similar and they could be normalized to avoid the duplicated code and query.

Now that I notice, they actually use the same statement name and neither of them check if it exists before preparing it.

Reply:
I have factored out the query and prepare statement so they are not repeated. I have added a check to see if the statement has previously been prepared. This will be in the next commit.